plumbing/protocol/packp/srvresp.go

```diff
@@ -1,97 +1,45 @@
 package packp
 
 import (
-	"bufio"
 	"bytes"
-	"errors"
 	"fmt"
 	"io"
 
 	"github.com/go-git/go-git/v5/plumbing"
 	"github.com/go-git/go-git/v5/plumbing/format/pktline"
-	"github.com/go-git/go-git/v5/plumbing/protocol/packp/capability"
-	"github.com/go-git/go-git/v5/utils/ioutil"
 )
 
 const ackLineLen = 44
 
 // ServerResponse object acknowledgement from upload-pack service
+// TODO: support multi_ack and multi_ack_detailed capabilities
 type ServerResponse struct {
 	ACKs []plumbing.Hash
-	req  *UploadPackRequest
 }
 
 // Decode decodes the response into the struct, isMultiACK should be true, if
 // the request was done with multi_ack or multi_ack_detailed capabilities.
-func (r *ServerResponse) Decode(reader io.Reader, isMultiACK bool) error {
-	s := bufio.NewReader(reader)
-
+func (r *ServerResponse) Decode(reader io.Reader) error {
 	var err error
 	for {
 		var p []byte
-		_, p, err = pktline.ReadLine(s)
+		_, p, err = pktline.ReadLine(reader)
 		if err != nil {
 			break
 		}
 
 		if err := r.decodeLine(p); err != nil {
 			return err
 		}
-
-		// we need to detect when the end of a response header and the beginning
-		// of a packfile header happened, some requests to the git daemon
-		// produces a duplicate ACK header even when multi_ack is not supported.
-		stop, err := r.stopReading(s)
-		if err != nil {
-			return err
-		}
-
-		if stop {
-			break
-		}
 	}
 
-	if errors.Is(err, io.EOF) {
-		return nil
+	if err == io.EOF {
```

Comment:
Do you mind elaborating on this change? AFAIK the recommended is use `errors.Is` over `err ==`.

Reply:
Good catch, must've been copied from `v5` prior to updating the go version